File: apps/java/design-pattern/src/main/java/pattern/strategy/PlainTextFormatter.java
package pattern.strategy;

import java.util.List;

/**
 * プレーンテキスト形式のフォーマッター
 */
public class PlainTextFormatter implements Formatter {

    @Override
    public String format(String title, List<String> text) {
        StringBuilder sb = new StringBuilder();
        sb.append("**** ").append(title).append(" ****\n");
        sb.append("\n");
        for (String line : text) {
            sb.append(line).append("\n");
        }
        return sb.toString();
    }
}

File: apps/php/design-pattern/src/Strategy/PlainTextFormatter.php
<?php

declare(strict_types=1);

namespace DesignPattern\Strategy;

class PlainTextFormatter
{
    public function __invoke(string $title, array $text): string
    {
        $lines = ["**** {$title} ****", ''];
        foreach ($text as $line) {
            $lines[] = $line;
        }
        return implode("\n", $lines);
    }
}
